### ISSUE TITLE
Iterate array is not reset after an unsuccessful time step

### ISSUE THREAD

**Issue by Yuriyzabegaev** [pmgbergen/porepy#1205]
The initial guess for the unknown solution is the previous time step solution. However, if Newton fails, and we use dynamic time stepper, the values in iterate are not reset to the previous time step values. This may lead to a resonance: We decrease the time step, but each time we start further and further from Newton's convergence region.

The fix would be to set these lines into `after_nonlinear_failure`:

```
prev_sol = self.equation_system.get_variable_values(time_step_index=0)
self.equation_system.set_variable_values(prev_sol, iterate_index=0)
```

I can do it if needed.

**Comment by IvarStefansson** [comment]
You have my blessing and thanks!

**Comment by keileg** [comment]
Resolved by #1201
